src/demo/state.js

```diff
@@ -16,10 +16,11 @@ export const getState = function() {
 
 export const setState = function(newState) {
   if (newState.currentMode) {
+    // When mode changes, reset backgroundImg and uiDrawn.
     state = {
-      ...initialState,
-      fishData: state.fishData,
-      trainer: state.trainer,
+      ...state,
+      backgroundImg: null,
+      uiDrawn: false,
```

Comment:
It seemed neat that the state functions were totally agnostic about our state.  Do they now need to know about app-specific options, or is that something the app code and perhaps the models can take care of?

Reply:
woops, i had this as a WIP and meant to remove it since i didn't end up using it. removing now!